idlc/src/main.rs

```diff
@@ -155,6 +159,11 @@ fn main() {
     trace!("Verifying interfaces");
     interface_verifier::InterfaceVerifier::new(&mir).run_pass();
 
+    let marking = idlc_codegen::documentation::Documentation::add_marking(
```

Comment:
You also don't need a special function to add marking. 

```rust
Documentation::new_with_idlc_doc(<MARKING>, DocumentationStyle::<LANGUAGE>);
```

should be good.

I also think we should rename this function and make it public, `idlc_doc` doesn't make sense. `new_from_raw_string` maybe?